site/src/pages/ManagementSettingsPage/OrganizationMembersPage.tsx

```diff
@@ -52,45 +54,96 @@ const OrganizationMembersPage: FC = () => {
-			canEditMembers={permissions.editMembers}
-			error={
-				membersQuery.error ??
-				addMemberMutation.error ??
-				removeMemberMutation.error ??
-				updateMemberRolesMutation.error
-			}
-			isAddingMember={addMemberMutation.isLoading}
-			isUpdatingMemberRoles={updateMemberRolesMutation.isLoading}
-			me={me}
-			members={members}
-			groupsByUserId={groupsByUserIdQuery.data}
-			addMember={async (user: User) => {
-				await addMemberMutation.mutateAsync(user.id);
-				void membersQuery.refetch();
-			}}
-			removeMember={async (member: OrganizationMemberWithUserData) => {
-				await removeMemberMutation.mutateAsync(member.user_id);
-				void membersQuery.refetch();
-			}}
-			updateMemberRoles={async (
-				member: OrganizationMemberWithUserData,
-				newRoles: string[],
-			) => {
-				await updateMemberRolesMutation.mutateAsync({
-					userId: member.user_id,
-					roles: newRoles,
-				});
-			}}
-		/>
+		<>
+			<OrganizationMembersPageView
+				allAvailableRoles={organizationRolesQuery.data}
+				canEditMembers={permissions.editMembers}
+				error={
+					membersQuery.error ??
+					addMemberMutation.error ??
+					removeMemberMutation.error ??
+					updateMemberRolesMutation.error
+				}
+				isAddingMember={addMemberMutation.isLoading}
+				isUpdatingMemberRoles={updateMemberRolesMutation.isLoading}
+				me={me}
+				members={members}
+				groupsByUserId={groupsByUserIdQuery.data}
+				addMember={async (user: User) => {
+					await addMemberMutation.mutateAsync(user.id);
+					void membersQuery.refetch();
+				}}
+				removeMember={setMemberToDelete}
+				updateMemberRoles={async (
+					member: OrganizationMemberWithUserData,
+					newRoles: string[],
+				) => {
+					await updateMemberRolesMutation.mutateAsync({
+						userId: member.user_id,
+						roles: newRoles,
+					});
+				}}
+			/>
+
+			<ConfirmDialog
+				type="delete"
+				open={memberToDelete !== undefined}
+				onClose={() => setMemberToDelete(undefined)}
+				title="Remove member"
+				confirmText="Remove"
+				onConfirm={async () => {
+					try {
+						if (memberToDelete) {
+							await removeMemberMutation.mutateAsync(memberToDelete?.user_id);
+						}
+						setMemberToDelete(undefined);
+						await membersQuery.refetch();
+						displaySuccess("User removed from organization successfully!");
+					} catch (error) {
+						setMemberToDelete(undefined);
+						displayError(
+							getErrorMessage(error, "Failed to remove user from organization"),
+						);
+					}
+				}}
+				description={
+					<Stack>
+						<p>
+							Removing this member will:
+							<ul>
+								<li>Remove the member from all groups in this organization</li>
+								<li>Remove all user role assignments</li>
+								<li>
+									Orphan all the member's workspaces associated with this
+									organization
+								</li>
+							</ul>
+						</p>
+
+						<p css={styles.test}>
+							Are you sure you want to remove{" "}
+							<strong>{memberToDelete?.username}</strong>?
```

Comment:
any way we could fiddle the types so that `memberToDelete` isn't optional here? seeing "Are you sure you want to remove ?" would be weird

Reply:
I feels difficult to remove the undefined from memberToDelete without restructuring the pattern that ConfirmDialog uses. For now, maybe its safer to just not display the username here.